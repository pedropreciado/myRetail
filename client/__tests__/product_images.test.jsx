import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Modal from 'react-modal';
import Renderer from 'react-test-renderer';

import ProductImages from '../components/product/show/product_images';

configure({ adapter: new Adapter() });

let images = [
  'image.jpg',
  'image_2.jpg',
  'image_3.jpg'
];

describe('ProductImages', () => {
  let productImages = shallow(
    <ProductImages 
    primaryImageIdx={2}
    images={images}
    />
  );

  it('renders correctly', () => {
    expect(productImages).toMatchSnapshot()
  });
  it('renders the primary image', () => {
    let primaryImage = productImages.find('#primary-image');
    expect(primaryImage.props().src).toBe('image_3.jpg');
  });
  it('begins with modal closed', () => {
    let modal = productImages.find(Modal);

    expect(modal.props().isOpen).toBe(false);
  });
  it('opens modal on selected image click', () => {
    let primaryImage = productImages.find('#primary-image');
    primaryImage.simulate('click');
    let modal = productImages.find(Modal);
    
    expect(modal.props().isOpen).toBe(true);
    modal.find('#in-modal').simulate('click'); // close modal
  });
  it('opens modal on "view larger" button', () => {
    let button = productImages.find('#zoom-image');
    
    button.simulate('click');
    let modal = productImages.find(Modal);

    expect(modal.props().isOpen).toBe(true);
    modal.find('#in-modal').simulate('click');
  });
});