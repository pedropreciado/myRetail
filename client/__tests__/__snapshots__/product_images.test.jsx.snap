// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImages renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductImages
    images={
        Array [
            "image.jpg",
            "image_2.jpg",
            "image_3.jpg",
          ]
    }
    primaryImageIdx={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="primary"
          id="primary-image"
          onClick={[Function]}
          src="image_3.jpg"
/>,
        <a
          id="zoom-image"
          onClick={[Function]}
>
          <FaSearchPlus />
          view larger
</a>,
        <ProductImageSlide
          changeImage={[Function]}
          currentImageIdx={2}
          images={
                    Array [
                              "image.jpg",
                              "image_2.jpg",
                              "image_3.jpg",
                            ]
          }
/>,
        <Modal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel=""
          isOpen={false}
          onRequestClose={[Function]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
>
          <img
                    alt="product"
                    id="in-modal"
                    onClick={[Function]}
                    src="image_3.jpg"
          />
</Modal>,
      ],
      "className": "product-image-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "primary",
          "id": "primary-image",
          "onClick": [Function],
          "src": "image_3.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FaSearchPlus />,
            "view larger",
          ],
          "id": "zoom-image",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "view larger",
        ],
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changeImage": [Function],
          "currentImageIdx": 2,
          "images": Array [
            "image.jpg",
            "image_2.jpg",
            "image_3.jpg",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ariaHideApp": false,
          "bodyOpenClassName": "ReactModal__Body--open",
          "children": <img
            alt="product"
            id="in-modal"
            onClick={[Function]}
            src="image_3.jpg"
/>,
          "closeTimeoutMS": 0,
          "contentLabel": "",
          "isOpen": false,
          "onRequestClose": [Function],
          "parentSelector": [Function],
          "portalClassName": "ReactModalPortal",
          "shouldCloseOnEsc": true,
          "shouldCloseOnOverlayClick": true,
          "shouldFocusAfterRender": true,
          "shouldReturnFocusAfterClose": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "product",
            "id": "in-modal",
            "onClick": [Function],
            "src": "image_3.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="primary"
            id="primary-image"
            onClick={[Function]}
            src="image_3.jpg"
/>,
          <a
            id="zoom-image"
            onClick={[Function]}
>
            <FaSearchPlus />
            view larger
</a>,
          <ProductImageSlide
            changeImage={[Function]}
            currentImageIdx={2}
            images={
                        Array [
                                    "image.jpg",
                                    "image_2.jpg",
                                    "image_3.jpg",
                                  ]
            }
/>,
          <Modal
            ariaHideApp={false}
            bodyOpenClassName="ReactModal__Body--open"
            closeTimeoutMS={0}
            contentLabel=""
            isOpen={false}
            onRequestClose={[Function]}
            parentSelector={[Function]}
            portalClassName="ReactModalPortal"
            shouldCloseOnEsc={true}
            shouldCloseOnOverlayClick={true}
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
>
            <img
                        alt="product"
                        id="in-modal"
                        onClick={[Function]}
                        src="image_3.jpg"
            />
</Modal>,
        ],
        "className": "product-image-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "primary",
            "id": "primary-image",
            "onClick": [Function],
            "src": "image_3.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FaSearchPlus />,
              "view larger",
            ],
            "id": "zoom-image",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "view larger",
          ],
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changeImage": [Function],
            "currentImageIdx": 2,
            "images": Array [
              "image.jpg",
              "image_2.jpg",
              "image_3.jpg",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ariaHideApp": false,
            "bodyOpenClassName": "ReactModal__Body--open",
            "children": <img
              alt="product"
              id="in-modal"
              onClick={[Function]}
              src="image_3.jpg"
/>,
            "closeTimeoutMS": 0,
            "contentLabel": "",
            "isOpen": false,
            "onRequestClose": [Function],
            "parentSelector": [Function],
            "portalClassName": "ReactModalPortal",
            "shouldCloseOnEsc": true,
            "shouldCloseOnOverlayClick": true,
            "shouldFocusAfterRender": true,
            "shouldReturnFocusAfterClose": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "product",
              "id": "in-modal",
              "onClick": [Function],
              "src": "image_3.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
